ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-composite-answer/activityCompositeAnswer.component.ts

```diff
@@ -42,38 +62,65 @@ export class ActivityCompositeAnswer implements OnChanges {
-                        // TODO investigate what is going on. Perhaps differences in TypeScript compiler target Javascript config?
-                        return rowOfAnswers.map((answerContainer: ActivityQuestionBlock<any>, index: number) =>
-                            this.buildBlockForChildQuestion(newBlock.children[index], answerContainer, newBlock.shown));
-                    });
-
-                    const blankRow: ActivityQuestionBlock<any>[] = this.block.children.map((questionBlock: ActivityQuestionBlock<any>) =>
-                        this.buildBlockForChildQuestion(questionBlock, null, this.block.shown));
-
-                    // If a row in Composite block has 2 and more questions,
-                    // and a user answered only on the first question and reloaded the page
-                    // the backend returns an array of answers only with 1 item,
-                    // so when the component will build rows, we will miss some blocks
-                    // example: https://broadinstitute.atlassian.net/browse/DDP-4536;
-                    // method below looks which blocks were missed and add them
-                    // Important: if a user answered only on the last question, backend returns the correct array of answers
-                    this.childQuestionBlocks = questionsRows.map((currentRow: ActivityQuestionBlock<any>[]) => {
-                        if (currentRow.length !== blankRow.length) {
-                            const missedQuestions = blankRow.slice(currentRow.length);
-                            currentRow.push(...missedQuestions);
-                        }
-                        return currentRow;
-                    });
+                    this.childQuestionBlocks = this.rebuildChildQuestions(newBlock, newAnswers);
+                }
+            }
+        }
+    }
+
+    private updateEquationQuestions(equations: AnswerResponseEquation[] = []): void {
+        for (const equation of equations) {
+            for (const question of this.block.children) {
+                if (question.stableId === equation.stableId) {
+                    question.setAnswer(equation.values, false);
+
+                    const prevAnswer: AnswerContainer[][] = this.block.answer;
+                    const newAnswer: AnswerContainer[][] = prevAnswer.map((answerRow: AnswerContainer[], index) =>
+                        // eslint-disable-next-line arrow-body-style
+                        answerRow.map((answer: AnswerContainer) => {
+                            return (answer.stableId === question.stableId) ? {
+                                ...answer,
+                                value: [equation.values[index]]
+                            } : answer;
+                        })
+                    );
+                    this.block.setAnswer(newAnswer, false);
+                    this.childQuestionBlocks = this.rebuildChildQuestions(this.block, newAnswer);
```

Comment:
Handling the server responses to  update equation here is definitely better than in the section component.
But could the `activityEquationAnswerComponent` itself subscribe to results from submissionmanager, see if its own stableid is referenced in the equation results and update itself if necessary? I don't think any other answer/question component cares, right?

Reply:
Right, other questions don't care.

Reply:
However, I can't see a decent way to update an equation only. 
Perhaps, just take a reference to a composite question(equation) and made force re-rendering?
Will try.

Reply:
Fixed in that^ way (https://github.com/broadinstitute/ddp-angular/pull/1222/commits/1dd4e07203e464d261e52c842a1e8371a890319a)